go-controller/pkg/node/gateway_iptables.go

```diff
@@ -394,6 +398,15 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []iptRule {
 			},
 			protocol: protocol,
 		},
+		{
+			table: "nat",
+			chain: "POSTROUTING",
+			args: []string{
+				"-s", masqueradeSubnet,
```

Comment:
I'm wondering if we should explicitly make this just the OVN masquerade IP. The reason is if someone tries to send traffic to a service from the host and lets say it used 169.254.169.2 IP on the host, then in zone 0 we would NAT this to primary IP of the interface I think. Then once the packet gets into breth0 it would hit:
```
 cookie=0xdeff105, duration=510.629s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
```

which would snat it back to 169.254.169.2 before going to OVN. Seems like doing these SNAT operations is just going to make things even harder to debug. Maybe it makes sense to keep things as minimal as possible in scope until we find out we actually need them.

Reply:
hmm yeah ack that's a good point. I'll start with `.1` then. I was being overly gracious haha...